stetho/src/main/java/com/facebook/stetho/inspector/network/ResponseBodyFileManager.java

```diff
@@ -83,4 +125,49 @@ public OutputStream openResponseBodyFile(String requestId, boolean base64Encode)
   private static String getFilename(String requestId) {
     return FILENAME_PREFIX + requestId;
   }
+
+  /**
+   * Associates an asynchronous pretty printer with a response request id
+   * The pretty printer will be used to pretty print the response body that has
+   * the particular request id
+   *
+   * @param requestId Unique identifier for the response
+   * as per {@link NetworkEventReporter.InspectorResponse#requestId()}
+   * @param asyncPrettyPrinter Asynchronous Pretty Printer to pretty print the response body
+   */
+  public void associateAsyncPrettyPrinterWithId(
+      String requestId,
+      AsyncPrettyPrinter asyncPrettyPrinter) {
+    if (mRequestIdMap.containsKey(requestId)) {
```

Comment:
This is not appropriately thread safe.  Whenever you require synchronization you must make sure all atomic operations are performed under the lock.  In this case, it would be possible for two threads to race associating the same async pretty printer with the request id and both would report that the key was not already there and then both would attempt to add it.

This is solved in one of two ways here: a) use a different locking object and synchronize on it for the entire method body or b) take advantage of the fact that `put` returns the old value if it was present and throw if it returns non-null.